<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>G4_56</class>
 <widget class="QWidget" name="G4_56">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>672</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>G4_56</string>
  </property>
  <widget class="QTextBrowser" name="textBrowser">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>20</y>
     <width>781</width>
     <height>151</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>幼圆</family>
     <pointsize>20</pointsize>
    </font>
   </property>
   <property name="html">
    <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;meta charset=&quot;utf-8&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'幼圆'; font-size:20pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'Microsoft YaHei UI'; font-size:9pt;&quot;&gt;题目：&lt;/span&gt;&lt;span style=&quot; font-family:'Microsoft YaHei UI'; font-size:10pt;&quot;&gt;采用&lt;/span&gt;&lt;span style=&quot; font-family:'宋体'; font-size:10.5pt;&quot;&gt;类继承的方法&lt;/span&gt;&lt;span style=&quot; font-family:'Calibri','sans-serif'; font-size:10.5pt;&quot;&gt;(&lt;/span&gt;&lt;span style=&quot; font-family:'宋体'; font-size:10.5pt;&quot;&gt;不直接修改类&lt;/span&gt;&lt;span style=&quot; font-family:'Calibri','sans-serif'; font-size:10.5pt;&quot;&gt;Student)&lt;/span&gt;&lt;span style=&quot; font-family:'宋体'; font-size:10.5pt;&quot;&gt;完成以下功能。&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'Wingdings'; font-size:10pt;&quot;&gt;l&lt;/span&gt;&lt;span style=&quot; font-family:'宋体'; font-size:10pt;&quot;&gt;建立一个名为&lt;/span&gt;&lt;span style=&quot; font-size:10pt;&quot;&gt;Student&lt;/span&gt;&lt;span style=&quot; font-family:'宋体'; font-size:10pt;&quot;&gt;的类，该类有以下几个私有数据成员：学生姓名、学号、性别和年龄。还有以下两个成员函数：一个用于初始化学生姓名、学号、性别和年龄的构造函数，一个用于输出学生信息的函数。&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'Wingdings'; font-size:10pt;&quot;&gt;l&lt;/span&gt;&lt;span style=&quot; font-family:'宋体'; font-size:10pt;&quot;&gt;增加以下私有成员变量：高等数学、英语、操作系统和数据结构&lt;/span&gt;&lt;span style=&quot; font-size:10pt;&quot;&gt;4&lt;/span&gt;&lt;span style=&quot; font-family:'宋体'; font-size:10pt;&quot;&gt;门课的分数和总成绩。修改或增加以下成员函数：初始化学生姓名、学号、性别、年龄和&lt;/span&gt;&lt;span style=&quot; font-size:10pt;&quot;&gt;4&lt;/span&gt;&lt;span style=&quot; font-family:'宋体'; font-size:10pt;&quot;&gt;门课分数的构造函数，输入&lt;/span&gt;&lt;span style=&quot; font-size:10pt;&quot;&gt;4&lt;/span&gt;&lt;span style=&quot; font-family:'宋体'; font-size:10pt;&quot;&gt;门课分数的函数，计算学生总成绩的函数，输出学生信息的函数。编写一个主函数，调用原来的构造函数声明一个学生对象并输入其&lt;/span&gt;&lt;span style=&quot; font-size:10pt;&quot;&gt;4&lt;/span&gt;&lt;span style=&quot; font-family:'宋体'; font-size:10pt;&quot;&gt;门课的分数，再调用新增加的构造函数声明另一个学生对象。最后分别计算两个学生的总分，并在屏幕输出&lt;/span&gt;&lt;span style=&quot; font-family:'宋体'; font-size:10pt; color:#ff0000;&quot;&gt;两个学生的所有信息。&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="id_1">
   <property name="geometry">
    <rect>
     <x>80</x>
     <y>272</y>
     <width>100</width>
     <height>25</height>
    </rect>
   </property>
   <property name="text">
    <string>1122334455</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QLabel" name="label_4">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>364</y>
     <width>41</width>
     <height>16</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>幼圆</family>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="text">
    <string>年龄：</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="age_1">
   <property name="geometry">
    <rect>
     <x>80</x>
     <y>360</y>
     <width>100</width>
     <height>25</height>
    </rect>
   </property>
   <property name="text">
    <string>18</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QLineEdit" name="gender_1">
   <property name="geometry">
    <rect>
     <x>80</x>
     <y>316</y>
     <width>100</width>
     <height>25</height>
    </rect>
   </property>
   <property name="text">
    <string>男</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QLabel" name="label_2">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>276</y>
     <width>41</width>
     <height>16</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>幼圆</family>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="text">
    <string>学号：</string>
   </property>
  </widget>
  <widget class="QPushButton" name="tian_jia">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>630</y>
     <width>151</width>
     <height>25</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>幼圆</family>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="text">
    <string>录 入 学 生 A</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_3">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>320</y>
     <width>41</width>
     <height>16</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>幼圆</family>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="text">
    <string>性别：</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="name_1">
   <property name="geometry">
    <rect>
     <x>80</x>
     <y>228</y>
     <width>100</width>
     <height>25</height>
    </rect>
   </property>
   <property name="text">
    <string>张三</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QLabel" name="label">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>232</y>
     <width>51</width>
     <height>16</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>幼圆</family>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="text">
    <string>姓名：</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_5">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>190</y>
     <width>161</width>
     <height>20</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>幼圆</family>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="text">
    <string>学  生  信  息</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QLabel" name="label_6">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>422</y>
     <width>161</width>
     <height>20</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>幼圆</family>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="text">
    <string>学  生  成  绩</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QLabel" name="label_10">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>464</y>
     <width>71</width>
     <height>16</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>幼圆</family>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="text">
    <string>高等数学：</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="math_1">
   <property name="geometry">
    <rect>
     <x>109</x>
     <y>460</y>
     <width>71</width>
     <height>25</height>
    </rect>
   </property>
   <property name="text">
    <string>0</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="Line" name="line">
   <property name="geometry">
    <rect>
     <x>15</x>
     <y>400</y>
     <width>181</width>
     <height>16</height>
    </rect>
   </property>
   <property name="orientation">
    <enum>Qt::Horizontal</enum>
   </property>
  </widget>
  <widget class="QLabel" name="label_11">
   <property name="geometry">
    <rect>
     <x>31</x>
     <y>504</y>
     <width>71</width>
     <height>16</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>幼圆</family>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="text">
    <string> 英  语：</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_12">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>590</y>
     <width>71</width>
     <height>16</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>幼圆</family>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="text">
    <string>数据结构：</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_13">
   <property name="geometry">
    <rect>
     <x>29</x>
     <y>550</y>
     <width>71</width>
     <height>16</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>幼圆</family>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="text">
    <string>操作系统：</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="caozuo_1">
   <property name="geometry">
    <rect>
     <x>108</x>
     <y>546</y>
     <width>71</width>
     <height>25</height>
    </rect>
   </property>
   <property name="text">
    <string>0</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QLineEdit" name="data_1">
   <property name="geometry">
    <rect>
     <x>108</x>
     <y>586</y>
     <width>71</width>
     <height>25</height>
    </rect>
   </property>
   <property name="text">
    <string>0</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QLineEdit" name="english_1">
   <property name="geometry">
    <rect>
     <x>108</x>
     <y>500</y>
     <width>71</width>
     <height>25</height>
    </rect>
   </property>
   <property name="text">
    <string>0</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QListView" name="xian_shi">
   <property name="geometry">
    <rect>
     <x>210</x>
     <y>190</y>
     <width>581</width>
     <height>441</height>
    </rect>
   </property>
  </widget>
  <widget class="QLabel" name="ti_shi">
   <property name="geometry">
    <rect>
     <x>210</x>
     <y>640</y>
     <width>221</width>
     <height>20</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>幼圆</family>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="text">
    <string/>
   </property>
   <property name="alignment">
    <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
   </property>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>tian_jia</sender>
   <signal>clicked()</signal>
   <receiver>G4_56</receiver>
   <slot>lu_ru()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>180</x>
     <y>640</y>
    </hint>
    <hint type="destinationlabel">
     <x>198</x>
     <y>642</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>lu_ru()</slot>
 </slots>
</ui>
